### PR TITLE
Feat add context variable fetching functionalities

### PR DESCRIPTION
**Pull request by alenmestrov**
Introduced endpoints for fetching context variables by key value or by limit and offset.

### CODE REVIEW

File: crates/context/config/src/client/env/proxy/query/context_storage_entries.rs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        let entries: Vec<(Box<[u8]>, Box<[u8]>)> = serde_json::from_slice(&response)
+            .map_err(|e| eyre::eyre!("Failed to decode response: {}", e))?;
+
+        // Convert to ContextStorageEntry
+        Ok(entries
+            .into_iter()
+            .map(|(key, value)| ContextStorageEntry {
+                key: key.to_vec(),
+                value: value.to_vec(),
+            })
```
Comment (miraclx) [inline crates/context/config/src/client/env/proxy/query/context_storage_entries.rs]: `[u8]::to_vec` copies the slice.

```suggestion
        let entries: Vec<(Vec<u8>, Vec<u8>)> = serde_json::from_slice(&response)
            .map_err(|e| eyre::eyre!("Failed to decode response: {}", e))?;

        // Convert to ContextStorageEntry
        Ok(entries
            .into_iter()
            .map(|(key, value)| ContextStorageEntry {
                key,
                value,
            })
```

or

```suggestion
        let entries: Vec<(Box<[u8]>, Box<[u8]>)> = serde_json::from_slice(&response)
            .map_err(|e| eyre::eyre!("Failed to decode response: {}", e))?;

        // Convert to ContextStorageEntry
        Ok(entries
            .into_iter()
            .map(|(key, value)| ContextStorageEntry {
                key: key.into(),
                value: value.into(),
            })
```

File: crates/context/config/src/client/env/proxy/query/context_storage_entries.rs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    fn encode(self) -> eyre::Result<Vec<u8>> {
+        // Encode offset and limit using Candid
+        Encode!(&self.offset, &self.limit)
+            .map_err(|e| eyre::eyre!("Failed to encode request: {}", e))
+    }
+
+    fn decode(response: Vec<u8>) -> eyre::Result<Self::Returns> {
+        // Decode the response as Vec of tuples
+        let entries: Vec<(Vec<u8>, Vec<u8>)> = Decode!(&response, Vec<(Vec<u8>, Vec<u8>)>)
```
Comment (miraclx) [inline crates/context/config/src/client/env/proxy/query/context_storage_entries.rs:96]: 👍🏽

**Review by miraclx — APPROVED** [review]

File: crates/server/src/admin/service.rs [hunk 3]
```diff
@@ -164,6 +165,14 @@ pub(crate) fn setup(
         .route(
             "/contexts/:context_id/proposals/:proposal_id",
             get(get_proposal_handler),
+        )
+        .route(
+            "/contexts/:context_id/proposals/get_context_value",
```
Comment (MatejVukosav) [inline crates/server/src/admin/service.rs]: Url should not have underscore

File: crates/context/config/src/client/env/proxy/query.rs [hunk 4]
```diff
@@ -73,4 +77,20 @@ impl<'a, T: Transport> ContextProxyQuery<'a, T> {
...
+
+        utils::send(&self.client, Operation::Read(params)).await
+    }
+
+    pub async fn get_context_storage_entries(
+        &self,
+        offset: usize,
+        limit: usize,
+    ) -> Result<Vec<ContextStorageEntry>, ClientError<T>> {
+        let params = ContextStorageEntriesRequest { offset, limit };
```
Comment (MatejVukosav) [inline crates/context/config/src/client/env/proxy/query.rs:92]: In description you mentioned size but here I can see only offset and limit?

File: crates/context/config/src/client/env/proxy/query/context_storage_entries.rs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+};
+use crate::client::env::Method;
+use crate::client::protocol::icp::Icp;
+use crate::client::protocol::near::Near;
+use crate::client::protocol::starknet::Starknet;
+use crate::types::ContextStorageEntry;
+
+#[derive(Clone, Debug, Serialize)]
+pub(super) struct ContextStorageEntriesRequest {
+    pub(super) offset: usize,
```
Comment (MatejVukosav) [inline crates/context/config/src/client/env/proxy/query/context_storage_entries.rs:17]: What is the use case for offset?

Comment (alenmestrov) [reply crates/context/config/src/client/env/proxy/query/context_storage_entries.rs:17]: pagination

File: crates/context/config/src/client/env/proxy/query.rs (hunk 4 above)
Comment (alenmestrov) [reply crates/context/config/src/client/env/proxy/query.rs:92]: updated description

**Pull request merged**